In [2]:
import pandas as pd
import numpy as np
import scipy.stats as ss
import statsmodels.api as sm

In [4]:
table = pd.read_csv("E:/XacSuatThongKe/DuLieuThucHanh/Du_lieu/hostel.csv")

##### **a. Viết mô hình hồi quy tuyến tính tổng thể biểu diễn valueformoney (Y) theo atmosphere (X)**
---

In [10]:
atmosphere = sm.add_constant(table.atmosphere)
model = sm.OLS(table.valueformoney, atmosphere).fit()
model.summary()

<class 'statsmodels.iolib.summary.Summary'>
"""
                            OLS Regression Results                            
==============================================================================
Dep. Variable:          valueformoney   R-squared:                       0.566
Model:                            OLS   Adj. R-squared:                  0.565
Method:                 Least Squares   F-statistic:                     386.1
Date:                Thu, 03 Oct 2024   Prob (F-statistic):           1.34e-55
Time:                        23:00:19   Log-Likelihood:                -300.73
No. Observations:                 298   AIC:                             605.5
Df Residuals:                     296   BIC:                             612.9
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
==============================================================================
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          3.8790      0.257     15.099      0.000       3.373       4.385
atmosphere     0.5995      0.031     19.650      0.000       0.539       0.659
==============================================================================
Omnibus:                       66.955   Durbin-Watson:                   2.013
Prob(Omnibus):                  0.000   Jarque-Bera (JB):              435.745
Skew:                          -0.714   Prob(JB):                     2.39e-95
Kurtosis:                       8.749   Cond. No.                         56.8
==============================================================================

Notes:
[1] Standard Errors assume that the covariance matrix of the errors is correctly specified.
"""

##### Mô hình hồi quy:  $$y = 3.8790 + 0.5995*x + \varepsilon$$
---

##### **b. Ước lượng mô hình hồi quy câu (a) (mô hình hồi quy mẫu) và nêu ý nghĩa của hệ số góc**

$\beta_1 = 0.5995$ - Ý nghĩa: Khi điểm đánh giá cho atmosphere tăng thêm 1 điểm thì trung bình cho valueformoney tăng thêm 0.5995 điểm

---

##### **d. Kiểm định tại mức ý nghĩa α = 5% xem biến atmosphere có ảnh hưởng có ý nghĩa đến biến valueformoney không?**

#### **$H_0: \beta_1 = 0$**

#### **$H_1: \beta_1 \ne 0$**

Cách 1:

p-giá trị là: P > |t| < 0.001 (vì kết quả cho đến 3 chữ số đằng sau dấu phẩy là 0 nên nhỏ hơn 0.001). Từ đó suy ra, p-giá trị < 0.05 nên bác bỏ H0 hay mô hình hồi quy có ý nghĩa. Tức là biến atmosphere ảnh hưởng có ý nghĩa đến biến valueformoney

Cách 2:

p-giá trị < 0.001 < 0.05. Từ đó suy ra bác bỏ H0. Tức là biến atmosphere có ảnh hưởng có ý nghĩa đến biến valueformoney

---

##### **e. Tìm khoảng tin cậy 90% cho hệ số góc của mô hình trong câu (a)**

---

In [24]:
model.conf_int(0.1)
# Khoảng tin cậy 90% cho hệ góc là [0.549122, 0.649795]

0         1
const       3.455119  4.302893
atmosphere  0.549122  0.649795

##### **f. Một nhà nghỉ có điểm đánh giá môi trường X = 9.1. Đưa ra giá trị dự báo cho điểm đánh giá số tiền bỏ ra xứng đáng valueformoney (Y) của nhà nghỉ này**

---

In [12]:
model.predict([1, 9.1])

array([9.33407714])

*Khi điểm đánh giá atmosphere là 9.1 thì điểm đanh giá valueformoney là 9.33*

---

##### **g. Tính hệ số xác định và nêu ý nghĩa**

---

**- Hệ số xác định của mô hình là $R^2 = R-squared = 0.566$**

*Mô hình phù hợp ở mức bình thường so với tập số liệu*

*56.6% sự biến động của điểm đánh giá valueformoney được giải thích qua điểm đánh giá atmosphere từ mô hình*

---

In [20]:
mu1 = table.atmosphere.mean()
mu2 = table.valueformoney.mean()
var1 = table.atmosphere.var()
var2 = table.valueformoney.var()
cov = np.cov(table.atmosphere, table.valueformoney)

print(mu1)
print(mu2)
print(var1)
print(var2)
print(cov)

8.325503355704697
8.86979865771812
1.6048692743994757
1.0188154475402798
[[1.60486927 0.96205229]
 [0.96205229 1.01881545]]
